components/event-bus/internal/knative/util/kn-eventing.go

```diff
@@ -245,6 +257,8 @@ func resendMessage(httpClient *http.Client, channel *evapisv1alpha1.Channel, hea
 	}
 	defer func() {
 		_ = res.Body.Close()
+		tran, _ := httpClient.Transport.(*http.Transport)
```

Comment:
If typecasting fails then this will panic. Let's use the `ok` pattern here.

Reply:
@sayanh Thanks for the catch, this was missed in the refactor, will fix it!